File: Java/2015/src/Day1.java
package src;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class Day1 {

	public static void main(String[] args) throws IOException {
		Part1();
		Part2();
	}
	
	public static void Part1() throws IOException {
		String input = Files.readString(Paths.get("2015/Input/Day1_Input.txt"));
		
		int floor = 0;
		for(char c : input.toCharArray()) {
			if(c == '(') floor++;
			else floor--;
		}
		System.out.println(floor);
	}
	
	public static void Part2() throws IOException {
		String input = Files.readString(Paths.get("2015/Input/Day1_Input.txt"));
		
		int floor = 0;
		for(int i = 0; i < input.length(); i++) {
			if(input.charAt(i) == '(') floor++;
			else floor--;
			
			if(floor == -1) {
				System.out.println(i+1);
				return;
			}
		}
	}

}

File: 2022/src/Day1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;

namespace Advent_of_Code.src
{
    internal class Day1
    {
        public static void Main(string[] args)
        {
            string[] input = File.ReadAllLines("../../../Input/Day1_Input.txt");
            Console.WriteLine(Part1(input));
            Console.WriteLine(Part2(input));
        }


        private static int Part1(string[] input)
        {
            int currentCalories = 0;
            int maxCalories = 0;
            foreach (string str in input)
            {
                if (str.Length == 0)
                {
                    if (currentCalories > maxCalories)
                    {
                        maxCalories = currentCalories;
                    }
                    currentCalories = 0;
                    continue;
                }

                int calories = int.Parse(str);
                currentCalories += calories;
            }
            return maxCalories;
        }

        private static int Part2(string[] input)
        {
            int currentCalories = 0;
            List<int> totalCalories = new List<int>();

            foreach (string str in input)
            {
                if (str.Length == 0)
                {
                    totalCalories.Add(currentCalories);
                    currentCalories = 0;
                    continue;
                }

                int calories = int.Parse(str);
                currentCalories += calories;
            }
            totalCalories.Sort();
            totalCalories.Reverse();

            return totalCalories[0] + totalCalories[1] + totalCalories[2];
        }
    }
}
